src/messaging/ExchangeContext.cpp

```diff
@@ -243,7 +243,7 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
 #if CONFIG_DEVICE_LAYER && CHIP_CONFIG_ENABLE_ICD_SERVER
             DeviceLayer::ChipDeviceEvent event;
             event.Type                       = DeviceLayer::DeviceEventType::kChipMsgSentEvent;
-            event.MessageSent.ExpectResponse = IsResponseExpected();
+            event.MessageSent.ExpectResponse = currentMessageExpectResponse;
```

Comment:
What does this event capture? Do we actually send back-to-back messages on the same exchange without expecting a response? This seems contrary to expected behavior.

Reply:
> Do we actually send back-to-back messages on the same exchange without expecting a response?

Consider the following message traffic:

1. A sends a message to B.
2. B sends response to A that expects a response from A.
3. A re-sends original message to B (if the response was lost, or just came later than the MRP resend timer).
4. B sends standalone ack for the duplicate.

If step 4 happens before the response to step 2 is received, then B is sending a message that expects a response, then on the same exchange sending a message (the standalone ack) that does not expect a response.

Reply:
Ok, Ya, I see the flow for sending a StandaloneAck. I presumed that this API was being used by the higher protocol layers only, and MRP control messages(StandaloneAcks) were internally handled and sent down the stack. In that case, this fix makes sense.